programs/drift-competitions/src/tests.rs

```diff
@@ -0,0 +1,299 @@
+#[cfg(test)]
+mod competition_fcn {
+    use crate::state::{Competition, CompetitionRoundStatus, Competitor};
+    use drift::{
+        math::{
+            constants::{PERCENTAGE_PRECISION, PERCENTAGE_PRECISION_U64, QUOTE_PRECISION},
+            insurance::if_shares_to_vault_amount,
+        },
+        state::{
+            insurance_fund_stake::InsuranceFundStake, spot_market::SpotMarket, user::UserStats,
+        },
+    };
+
+    #[test]
+    fn test_competition_settlement() {
+        let mut now = 168000000;
+        let sweepstakes = &mut Competition::default();
+
+        sweepstakes.first_round_expiry_ts = now + 60;
+        sweepstakes.round_duration = 60;
+        assert_eq!(sweepstakes.status, CompetitionRoundStatus::Active);
+
+        sweepstakes.number_of_competitors = 2;
+        let comp1 = &mut Competitor::default();
+        comp1.claim_entry().unwrap();
+
+        let comp2 = &mut Competitor::default();
+        comp2.claim_entry().unwrap();
+        comp2.bonus_score += 2;
+
+        let us: &UserStats = &UserStats::default();
+
+        assert!(sweepstakes.settle_competitor(comp1, us, now).is_err());
+        assert!(sweepstakes.settle_competitor(comp2, us, now).is_err());
+
+        now += 60;
+
+        assert!(sweepstakes.reset_round().is_err());
+        assert!(sweepstakes.resolve_winning_draw().is_err());
+
+        sweepstakes.settle_competitor(comp1, us, now).unwrap();
+        assert!(sweepstakes.number_of_competitors_settled == 1);
+        sweepstakes.settle_competitor(comp1, us, now).unwrap();
+        assert!(sweepstakes.number_of_competitors_settled == 1); // resettle a competitor doesnt increment
+
+        sweepstakes.settle_competitor(comp2, us, now).unwrap();
+        assert!(sweepstakes.number_of_competitors_settled == 2);
```

Comment:
these tests might be safer if you assert the whole `swepstakes` equals a `expected_sweepstakes` where you declare the whole thing, that way you know you assert every update/dont forget any?